src/staticdata.c

```diff
@@ -198,6 +198,20 @@ static uintptr_t jl_fptr_id(void *fptr)
         return *(uintptr_t*)pbp;
 }
 
+int32_t jl_jlcall_api(const char *fname)
+{
+    // give the function an index in the constant lookup table
+    if (fname == NULL)
+        return 0;
+    if (!strncmp(fname, "japi3_", 6)) // jlcall abi 3 from JIT
+        return JL_API_WITH_PARAMETERS;
+    assert(!strncmp(fname, "japi1_", 6) ||  // jlcall abi 1 from JIT
+           !strncmp(fname, "jsys1_", 6) ||  // jlcall abi 1 from sysimg
+           !strncmp(fname, "jlcall_", 7) || // jlcall abi 1 from JIT wrapping a specsig method
+           !strncmp(fname, "jlsysw_", 7));  // jlcall abi 1 from sysimg wrapping a specsig method
+    return JL_API_GENERIC;
+}
```

Comment:
This isn't static data, it's compiler.cpp data. Why did this move?

Reply:
For builds that exclude codegen. It's such a simple function we ought to be able to use it without llvm. I put it here because this file uses it.

Reply:
For compiles that don't have a JIT, why would they care what the JIT would do if it was present?

Reply:
Fair enough; my only goal here is to solve a linking problem. Another solution might be to separate reading and writing staticdata, since all builds need to read it but only builds with codegen need to write it. That would be more disruptive though.